README.md

```diff
@@ -1,480 +1,479 @@
-  - `telescope.builtin.lsp_workspace_symbols` and `telescope.builtin.lsp_document_symbols` are also good to bind for LSP.
-
-## Mappings
+To embed the above code snippet in a `.vim` file
+  (for example in `after/plugin/telescope.nvim.vim`),
+  wrap it in `lua << EOF code-snippet EOF`:
 
-Mappings are fully customizable. Many familiar mapping patterns are setup as defaults.
-
-```
-<C-n>  <C-p> next | previous
-<Down> <Up>  next | previous
-j      k     next | previous (in normal mode)
-<CR>         go to file selection
-
-<C-x>        go to file selection as a split
-<C-v>        go to file selection as a vertical split
-<C-t>        go to a file in a new tab
-
-<C-u>        scroll up in preview window
-<C-d>        scroll down in preview window
-
-<C-c>        close telescope
-<Esc>        close telescope (in normal mode)
+```lua
+lua << EOF
+require('telescope').setup{
+  -- ...
+}
+EOF
 ```
 
-To see the full list of mappings, check out `lua/telescope/mappings.lua` and the `default_mappings` table.
-
-To override ALL of the default mappings, you can use the `default_mappings` key in the `setup` table.
-
+<!-- TODO: move some options to Options affecting Behaviour -->
+
+#### Options affecting Presentation
+
+| Keys                   | Description                                           | Options                    |
+|------------------------|-------------------------------------------------------|----------------------------|
+| `prompt_position`      | Where the prompt should be located.                   | top/bottom                 |
+| `prompt_prefix`        | What should the prompt prefix be.                     | string                     |
+| `sorting_strategy`     | Where first selection should be located.              | descending/ascending       |
+| `layout_strategy`      | How the telescope is drawn.                           | [supported layouts](https://github.com/nvim-telescope/telescope.nvim/wiki/Layouts) |
+| `winblend`             | How transparent is the telescope window should be.    | NUM                        |
+| `layout_defaults`      | Layout specific configuration ........ TODO           | TODO                       |
+| `width`                | TODO                                                  | NUM                        |
+| `preview_cutoff`       | TODO                                                  | NUM                        |
+| `results_height`       | TODO                                                  | NUM                        |
+| `results_width`        | TODO                                                  | NUM                        |
+| `borderchars`          | The border chars, it gives border telescope window    | dict                       | 
+| `color_devicons`       | Whether to color devicons or not                      | boolean                    |
+| `use_less`             | Whether to use less of cat/bat                        | boolean                    |
```

Comment:
> Whether to use less of cat/bat

Isn't this supposed to be "use less or cat/bat" ? Didn't notice this earlier.

Reply:
@Conni2461 I think less apply to both cat and bat, no?

Reply:
Okay i just tested it. Use less is only used for bat. It only sets an option.
https://github.com/nvim-telescope/telescope.nvim/blob/e16712f22e7702cde804f66a81437eaae5ec7d1b/lua/telescope/previewers.lua#L40-L50
But we can also use less if only cat is installed, so people without bat can also scroll. Someone interested in updating this, if not i will provide a PR in the next couple of days.

Or we just move completely to vim buffers, like the current previewer for tags. But we than have to implement scroll for vim buffers.

Opinions?

Reply:
@Conni2461 the preview for tags feels really light (no more flickering when switch between results for the same file) and highlights correctly too.  I think vim buffers should be used for everything else, don't you think @tjdevries?. 

I believe anything that let us to replace external dependencies will always be better for performance and stability .

Reply:
If I'm not mistaken, TJ wants to be able to use treesitter highlights for file preview so we won't need bat but that buffer will only have treesitter attached and completely nothing else (saw the discussion on gitter a whilte ago, correct me if I'm wrong). 

I think for the time being, we don't need vim buffer. It will become slow right? Because it loads autocmd and other stuff for that filetype. So using `cat/bat` + `less` is fine IMO.

Reply:
hmmm @elianiva, valid point about autcmds but needs some testing . I'm not big fan of TS yet as it  keeps breaking my highlights, but perhaps in far feature it can be a replacement for vim buffers if proven stable.

Reply:
Open a discussion here https://github.com/nvim-telescope/telescope.nvim/issues/257

Reply:
@elianiva i know about that open discussion with "too many autocmd", thats why vim buffer for tags preview is more like an work in progress so people can report how it works. 

We will also probably move to a vim buffer for `telescope.builtin`.buffers because these buffers are usually already loaded with all highlighting, etc and we can just display them in the preview window. For unloaded but listed buffers, we just have to load them once, that similar to what we are doing in tags. We have to load a buffer once and then can just "reuse" it. You can check out #208. It is already implemented there.